src/main/java/com/rcv/StreamingCVRReader.java

```diff
@@ -0,0 +1,340 @@
+      // function: startRow
+      // purpose: startRow callback handler during xml parsing
+      // param: i the row which has started
+      @Override
+      public void startRow(int i) {
+        // we assume exactly one header row so skip the first row
+        if(i > 0) {
+          beginCVR();
+        }
+      }
+
+      // function: endRow
+      // purpose: endRow callback handler during xml parsing
+      // row has completed, we will create a new cvr object
+      // param: i the row which has ended
+      @Override
+      public void endRow(int i) {
+        // we assume exactly one header row so skip the first row
+        if(i > 0) {
+          endCVR();
+        }
+      }
+
+      // function: cell
+      // purpose: cell callback handler during xml parsing
+      // param: s cell address encoded as col,row
+      // param: s1 cell data
+      // param: xssfComment additional cell data (unused)
+      @Override
+      public void cell(String s, String s1, XSSFComment xssfComment) {
+        // address contains the row and col of this cell
+        Pair<Integer,Integer> address = getCellAddress(s);
+        int col = address.getKey();
+        int row = address.getValue();
+        // we assume exactly one header row so skip cells in the first row
+        if(row > 0) {
+          CVRCell(col, row, s1);
+        }
+      }
+
+      // function: headerFooter
+      // purpose: header footer callback from xml parsing - unused
+      // param: s header footer data
+      // param: b header footer data
+      // param: s1 header footer data
+      @Override
+      public void headerFooter(String s, boolean b, String s1) {
+        Logger.log(Level.WARNING, String.format("unexpected xml data: %s %d %s", s, b, s1));
+      }
+    };
+
+    // create the ContentHandler to handle parsing callbacks
+    ContentHandler handler = new XSSFSheetXMLHandler(styles,
+        sharedStrings,
+        sheetContentsHandler,
+        true);
+
+    // create the xml parser and set content handler
+    XMLReader parser = XMLReaderFactory.createXMLReader();
+    parser.setContentHandler(handler);
+    // trigger parsing
+    parser.parse(new InputSource(reader.getSheetsData().next()));
+
+    // throw if there were any unrecognized candidates -- this is considered bad
+    if (this.unrecognizedCandidateCounts.size() > 0) {
+      throw new UnrecognizedCandidatesException(unrecognizedCandidateCounts);
+    }
+
+    // return the input list with additions
+    return CVRList;
+  }
+
+  // exception class used when an unrecognized candidate is encountered during cvr parsing
+  // purpose is to help identify issues with cast vote record files and configuration files
+  static class UnrecognizedCandidatesException extends Exception {
+
+    // candidateCounts maps an unrecognized candidate name to the count of how many times it was
+    // encountered during CVR parsing
+    public final Map<String, Integer> candidateCounts;
```

Comment:
I'm curious about this. My general understanding was that it was considered good practice to make instance variables private and provide accessors when necessary. Do you think that's unnecessary?

Reply:
In cases like this where the object is just a container for data I do not bother, in the interest of simpler code.  It's not a hill I want to die on.